tests/conftest.py

```diff
@@ -172,11 +174,41 @@ def bdb_port():
     return environ.get('BDB_PORT', '9984')
 
 
+@fixture
+def bdb_node_header():
+    return {'app_id': 'id'}
+
+
 @fixture
 def bdb_node(bdb_host, bdb_port):
     return 'http://{host}:{port}'.format(host=bdb_host, port=bdb_port)
 
 
+@fixture
+def bdb_nodes_headers(bdb_node, bdb_node_header):
+    return [
+        {'endpoint': 'bigchaindb-driver_bigchaindb_2:9984',
```

Comment:
Do the new tests require all these 5 instances to be up and running? How do you assert which endpoint was actually called?

Reply:
> After that i stop-start 1 to 3 nodes nodes when the simulation (tests) is running to how see how the tests go.

This is great and definitely something worth doing! On the other hand, it is difficult to automate. So a better alternative for the automated test suite, like this one, is to mock the basic request method (e. g. `connection.request`) to behave differently depending on the URL passed, what allows to make all of the important assertions.

> main reason for the sleep(async wait)

`sleep` is an expensive approach to this problem. A better way is to retry "not found" requests - then you get closer to the time you actually need to spend to receive certain data. Off course, if you mock the request function, you do not have this problem at all.